src/iocp/mod.rs

```diff
@@ -442,6 +447,27 @@ impl Events {
     }
 }
 
+/// A packet used to wake up the poller with an event.
+#[derive(Debug, Clone)]
+pub struct CompletionPacket(Packet);
+
+impl CompletionPacket {
+    /// Create a new completion packet with a custom event.
+    pub fn new(event: Event) -> Self {
+        Self(Arc::pin(IoStatusBlock::from(PacketInner::Custom { event })))
+    }
+
+    /// Get the event associated with this packet.
+    pub fn event(&self) -> Event {
+        let data = self.0.as_ref().data().project_ref();
+
+        match data {
+            PacketInnerProj::Custom { event } => *event,
```

Comment:
why do we copy the event here and don't return a reference?

Reply:
Because `Event` is a `Copy` struct... although it might not be in the future. Fixed.